pkg/database/plugin.go

```diff
@@ -350,6 +350,11 @@ type iConfigRecordCollection interface {
 	DeleteConfigRecord(ctx context.Context, key string) (err error)
 }
 
+type iTokenPoolCollection interface {
+	// UpsertTokenPool - Upsert a token pool
+	UpsertTokenPool(ctx context.Context, data *fftypes.TokenPool, allowExisting bool) (err error)
```

Comment:
We will need the getters too. You might find it easiest to just copy/paste those all over from another collection and pop them in at the start (that's what we've done for adding other collections).

Reply:
Yea, was just picking a single thin thread since it already ended up as 1500+ LoC. But I can look at adding the getters if you think they should go in with the first push.

Reply:
I went ahead and pushed up some of the basic work for database getters, along with a very basic E2E database test.